packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -118,3 +118,33 @@ function getNodeCompatEntries() {
 
 	return entries;
 }
+
+/**
+ * Discerns wether workerd would populate the `process.env` object a worker or not
+ *
+ * @param workerConfig the worker configuration
+ * @returns true if workerd would populate the `process.env` given the provided configuration, false otherwise
+ */
+export function wouldWorkerdPopulateProcessEnv(workerConfig: WorkerConfig) {
+	const workerCompatFlags = workerConfig.compatibility_flags;
+
+	if (!workerCompatFlags.includes("nodejs_compat")) {
+		return false;
+	}
+
+	if (workerCompatFlags.includes("nodejs_compat_populate_process_env")) {
+		return true;
+	}
+
+	const compatDateEnablingProcessEnvPopulation = new Date("2025-04-01");
```

Comment:
Note: I couldn't add playground/e2e tests for this scenario, because we get a miniflare error when a future date is being used:
![Screenshot 2025-03-14 at 16 36 16](https://github.com/user-attachments/assets/f10d78ed-5a32-4ce5-96de-e1167376f055)

I'll create an issue to add these tests after the first or april if that's ok with everyone

Reply:
No longer applicable since we decided to just use `isNodeCompat` instead of relying on the compat date and flag directly